### PR TITLE
Commerce 12776

### PR DESCRIPTION
**Pull request by andrea-ale-sbarra**

### CODE REVIEW

File: modules/apps/commerce/commerce-product-type-virtual-web/src/main/resources/META-INF/resources/edit_commerce_virtual_order_item_file_entry.jsp [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+		<aui:button-row>
+			<aui:button cssClass="btn-lg" type="submit" value="save" />
+
+			<aui:button cssClass="btn-lg" type="cancel" />
+		</aui:button-row>
+	</aui:form>
+</liferay-frontend:side-panel-content>
+
+<aui:script sandbox="<%= true %>">
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-type-virtual-web/src/main/resources/META-INF/resources/edit_commerce_virtual_order_item_file_entry.jsp:80]: Can we move this into a js file?

File: modules/apps/commerce/commerce-product-type-virtual-web/src/main/resources/META-INF/resources/edit_cpd_virtual_setting_file_entry.jsp [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+		<aui:button-row>
+			<aui:button cssClass="btn-lg" type="submit" value="save" />
+
+			<aui:button cssClass="btn-lg" type="cancel" />
+		</aui:button-row>
+	</aui:form>
+</liferay-frontend:side-panel-content>
+
+<aui:script sandbox="<%= true %>">
```
Comment (alessiorendina) [inline modules/apps/commerce/commerce-product-type-virtual-web/src/main/resources/META-INF/resources/edit_cpd_virtual_setting_file_entry.jsp:85]: Can we move this into a js file?

**Review by alessiorendina — CHANGES_REQUESTED** [review]
@andrea-ale-sbarra Left some comments for you, thanks!

File: modules/apps/commerce/commerce-product-type-virtual-web/src/main/java/com/liferay/commerce/product/type/virtual/web/internal/frontend/view/table/CPDefinitionVirtualSettingTableFDSView.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+		FDSTableSchemaBuilder fdsTableSchemaBuilder =
+			_fdsTableSchemaBuilderFactory.create();
+
+		return fdsTableSchemaBuilder.add(
+			"cpDefinitionVirtualSettingFileId",
+			"cpDefinitionVirtualSettingFileId"
+		).add(
+			"fileEntryId", "fileEntryId"
+		).add(
+			"url", "url"
```
[the remark this answers is not in the document]
Comment (andrea-ale-sbarra) [reply modules/apps/commerce/commerce-product-type-virtual-web/src/main/java/com/liferay/commerce/product/type/virtual/web/internal/frontend/view/table/CPDefinitionVirtualSettingTableFDSView.java:39]: I asked David about this, he said that the URL makes sense

File: modules/apps/commerce/commerce-media-impl/src/main/java/com/liferay/commerce/media/internal/servlet/CommerceMediaServlet.java [hunk 4]
```diff
@@ -301,7 +303,19 @@ private void _sendMediaBytes(
...
+							commerceVirtualOrderItem.
+								getCommerceVirtualOrderItemFileEntries(),
+							commerceVirtualOrderItemFileEntry ->
+								commerceVirtualOrderItemFileEntry.
+									getCommerceVirtualOrderItemFileEntryId()),
+						fileEntryId)) {
+
+					_sendError(
+						httpServletResponse, HttpServletResponse.SC_NOT_FOUND,
+						"The file entry " + fileEntryId + " does not exist");
```
[the remark this answers is not in the document]
Comment (andrea-ale-sbarra) [reply modules/apps/commerce/commerce-media-impl/src/main/java/com/liferay/commerce/media/internal/servlet/CommerceMediaServlet.java:317]: This message is already in, I've just reused what was already in.

File: modules/apps/commerce/commerce-media-impl/src/main/java/com/liferay/commerce/media/internal/servlet/CommerceMediaServlet.java [hunk 5]
```diff
@@ -549,19 +563,35 @@ private void _sendVirtualSettingsMediaBytes(
 
 			if (sample) {
 				fileEntry = cpDefinitionVirtualSetting.getSampleFileEntry();
+
+				if ((fileEntry == null) ||
+					(fileEntry.getFileEntryId() != fileEntryId)) {
+
+					_sendError(
+						httpServletResponse, HttpServletResponse.SC_NOT_FOUND,
+						"The file entry " + fileEntryId + " does not exist");
```
[the remark this answers is not in the document]
Comment (andrea-ale-sbarra) [reply modules/apps/commerce/commerce-media-impl/src/main/java/com/liferay/commerce/media/internal/servlet/CommerceMediaServlet.java:570]: This message is already in, I've just reused what was already in.

File: modules/apps/commerce/commerce-product-type-virtual-api/bnd.bnd [hunk 6]
```diff
@@ -1,6 +1,6 @@
 Bundle-Name: Liferay Commerce Product Type Virtual API
 Bundle-SymbolicName: com.liferay.commerce.product.type.virtual.api
-Bundle-Version: 8.0.2
+Bundle-Version: 9.0.0
```
[the remark this answers is not in the document]
Comment (andrea-ale-sbarra) [reply modules/apps/commerce/commerce-product-type-virtual-api/bnd.bnd:3]: sure thing, I'll make a new PR out of this writing a new history..

File: modules/apps/commerce/commerce-product-type-virtual-order-api/bnd.bnd [hunk 7]
```diff
@@ -1,6 +1,6 @@
 Bundle-Name: Liferay Commerce Product Type Virtual Order API
 Bundle-SymbolicName: com.liferay.commerce.product.type.virtual.order.api
-Bundle-Version: 8.3.2
+Bundle-Version: 9.0.0
```
[the remark this answers is not in the document]
Comment (andrea-ale-sbarra) [reply modules/apps/commerce/commerce-product-type-virtual-order-api/bnd.bnd:3]: sure thing, I'll make a new PR out of this writing a new history..

**Pull request closed without merging**